modern/vite.config.js

```diff
@@ -19,8 +19,11 @@ export default defineConfig(() => ({
     svgr(),
     react(),
     VitePWA({
+      includeAssets: ['favicon.ico', 'apple-touch-icon-180x180.png'],
       workbox: {
         navigateFallbackDenylist: [/^\/api/],
+        maximumFileSizeToCacheInBytes: 4000000,
+        globPatterns: ['**/*.{js,css,html,woff,woff2,mp3}'],
```

Comment:
Can you please explain each of the change.

Reply:
By default, Vite PWA generates a Service Worker that caches every `.{js,css,html}` file in `assets/` folder that is smaller than 2MB. 
`globPatterns` change here, extends the extensions to woff font files and mp3 for the alarm sound. 
`maximumFileSizeToCacheInBytes` increases the size limit to 4MB, because Traccars main generated JS file (index-hash.js) is now bigger than 2MB. 
`includeAssets` includes the favicon.ico and apple-touch-icon static files that are not in `assets/` folder.

All this 3 changes combined, make sure that all the static assets that we need for a working PWA are cached by the Service Worker. This then means that the Service Worker serves them, even before a HTTP browser cache is involved. 
![image](https://github.com/traccar/traccar-web/assets/858821/eeacc044-802d-4567-9fe3-40e2287944bb)
Source: https://web.dev/articles/service-worker-caching-and-http-caching

Before this changes (and in recent builds) Service Worker only cached `index.html` and some internal .js files that it needed for work - which caused a problem if you loaded the PWA at least once (the Service Worker got installed and it cached the `index.html`), then you opened the PWA (or refreshed the page) without internet connection. The `index.html` got served from the Service Worker, but the `index-hash.js` HTTP request failed -> and this resulted in a blank white webpage with no indicator that anything is wrong. So with ensuring that all the assets are precached with the Service Worker version together, we avoid problems like this (or partial upgrades, where Service Worker serves cached `index.html` which is referencing now deleted `index-hash.js` file, that might not be in browser cache anymore - before the new Service Worker with new `index-hash.js` reference is installed/updated).

This also prepares the groundwork for possible offline functionality - as now the Service Worker (after first load) caches all the assets (html,css,fonts,...) we need for the PWA to work. And maybe we can, down the line, show something more than the "there is no network" error that is shown after this PR (instead of a blank page).

Reply:
This is awesome. That's for looking into this. What about other files outside of the `assets/` folder? Like all of the PWA images.

Reply:
Per the Vite PWA documentation those are included automatically:
`You don't need to configure PWA Manifest icons on includeAssets option`
Source: https://vite-pwa-org.netlify.app/guide/static-assets.html
The same goes for the base `styles.css`, it's not mentioned in the docs, but I can confirm they get included automatically too:
![image](https://github.com/traccar/traccar-web/assets/858821/b96318d1-d899-4523-9080-fb4811f1ad57)

So after another pass, I think Service Worker now caches all static assets it can.